packages/truffle-config/index.js

```diff
@@ -67,6 +67,9 @@ function Config(truffle_directory, working_directory, network) {
     }
   };
 
+  const resolveDirectory = (value) =>
+    path.resolve(self.working_directory, value);
```

Comment:
I don't really know a good place for this. I was going to avoid abstracting it, but I didn't like copying and pasting it everywhere.

Thoughts?

Reply:
I like it abstracted -- makes it clearer what's happening. I wonder if there is any way to move this logic into the addProp function, or would that just make things messier? I'm just thinking since you are already checking there whether you need to account for the path anyway?

Reply:
The reason it's not part of `addProp` is because certain properties are directories, and that feels like it should be explicitly indicated as such. If I were to write this from scratch, I'd define a bunch of helper methods for various data types, and `directory` or (`path`) might be one of them.